src/tokio.rs

```diff
@@ -69,54 +69,56 @@ impl io::Write for NetlinkSocket {
     }
 }
 
-impl Io for NetlinkSocket {
-    fn poll_read(&mut self) -> Async<()> {
-        <NetlinkSocket>::poll_read(self)
-    }
+pub struct NetlinkCodec {}
 
-    fn poll_write(&mut self) -> Async<()> {
-        <NetlinkSocket>::poll_write(self)
-    }
+impl tokio_io::AsyncRead for NetlinkSocket {
 }
 
-struct NetlinkCodec {
-
+impl tokio_io::AsyncWrite for NetlinkSocket {
+    fn shutdown(&mut self) -> Poll<(), io::Error> {
+        Ok(().into())
+    }
 }
 
-impl Codec for NetlinkCodec {
-    type In = NetlinkPacket<'static>;
-    type Out = NetlinkPacket<'static>;
 
-    fn decode_eof(&mut self, buf: &mut EasyBuf) -> io::Result<Self::In> {
+impl tokio_io::codec::Decoder for NetlinkCodec {
+    type Item = NetlinkPacket<'static>;
+    type Error = io::Error;
+
+    fn decode_eof(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
         println!("DECODE EOF CALLED");
 
-        Ok(NetlinkPacket::owned(buf.as_slice().to_owned()).unwrap())
+        Ok(NetlinkPacket::owned(buf[..].to_owned()))
     }
 
-    fn decode(&mut self, buf: &mut EasyBuf) -> io::Result<Option<Self::In>> {
+    fn decode(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
         let (owned_pkt, len) = {
-            let slice = buf.as_slice();
-            if slice.len() == 0 {
+            if buf.len() == 0 {
                 return Ok(None);
             }
-            if let Some(pkt) = NetlinkPacket::new(slice) {
-                //println!("{:?} slice: {}", pkt, slice.len());
+            if let Some(pkt) = NetlinkPacket::new(buf) {
```

Comment:
Maybe use an expect here.

Reply:
I think it's better not to panic, since it's less trouble for consumers. I've amended with an Err return for this case, and also removed the println calls.